Assembler.Base/BaseFrameHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using Assembler.Core;
+using Assembler.Core.Entities;
+
+namespace Assembler.Base
+{
+    public abstract class BaseFrameHandler<TFrame, TMessage> : IFrameHandler<TFrame, TMessage>
+        where TFrame : BaseFrame
+        where TMessage : BaseMessageInAssembly
+    {
+        protected readonly ITimeBasedCache<TMessage> Cache;
+        protected readonly IFactory<TFrame, string> IdentifierFactory;
+        protected readonly IMessageEnricher<TFrame, TMessage> MessageEnricher;
+        protected readonly ICreator<TMessage> MessageInAssemblyCreator;
+
```

Comment:
What about holding protected Logger here?

I think that running `_logger = loggerFactory.GetLogger(this);` in a base class should work and send the lowest implementation in the heirarchy

Reply:
It does not by what I remember, we tried it with the new Rabbit and it didn't work, we need a logger on the lower implementation

Reply:
use `this.GetType().Name` (You need to update ILoggerFactory interface with a another GetLogger function that receive string)

https://stackoverflow.com/questions/972494/from-base-class-in-c-get-derived-type

Reply:
I fixed it in the logger of Blizzard, would change the code everywhere

Reply:
I saw you fixed, but I didnt understand your comment.

Reply:
I'm using a different logger now